pyvista/core/utilities/geometric_objects.py

```diff
@@ -363,6 +364,353 @@ def Sphere(
     return surf
 
 
+def SolidSphere(
+    outer_radius=0.5,
+    inner_radius=0.0,
+    radius_resolution=5,
+    start_theta=0.0,
+    end_theta=360.0,
+    theta_resolution=30,
+    start_phi=0.0,
+    end_phi=180.0,
+    phi_resolution=30,
+    radius=None,
+    theta=None,
+    phi=None,
+    center=(0.0, 0.0, 0.0),
+    direction=(0.0, 0.0, 1.0),
+):
+    """Create a solid sphere.
+
+    Parameters
+    ----------
+    outer_radius : float, default: 0.5
+        Outer radius of sphere.
+
+    inner_radius : float, default: 0.0
+        Inner radius of sphere.
+
+    radius_resolution : int, default: 5
+        Number of points in radial direction.
+
+    start_theta : float, default: 0.0
+        Starting longitude angle in degrees.
+
+    end_theta : float, default: 360.0
+        Ending longitude angle in degrees.
+
+    theta_resolution : int, default: 30
+        Number of points in theta direction.
+
+    start_phi : float, default: 0.0
+        Starting latitude angle in degrees.
+
+    end_phi : float, default: 180.0
+        Ending latitude angle in degrees.
+
+    phi_resolution : int, default: 30
+        Number of points in phi direction,
+        inclusive of polar axis if applicable.
+
+    radius : sequence[float], optional
+        A monotonically increasing sequence of values specifying radial
+        points. If specified, it overrides any choices for ``outer_radius``,
+        ``inner_radius``, and ``radius_resolution``.
+
+    theta : sequence[float], optional
+        A monotonically increasing sequence of values specifying theta
+        points in degrees. If specified, it overrides any choices for
+        ``start_theta``, ``end_theta``, and ``theta_resolution``.
+
+    phi : sequence[float], optional
+        A monotonically increasing sequence of values specifying phi
+        points in degrees. If specified, it overrides any choices for
+        ``start_phi``, ``end_phi``, and ``phi_resolution``.
+
+    center : sequence[float], default: (0.0, 0.0, 0.0)
+        Center in ``[x, y, z]``.
+
+    direction : sequence[float], default: (0.0, 0.0, 1.0)
+        Direction vector in ``[x, y, z]`` pointing from ``center`` to
+        the sphere's north pole at zero degrees latitude.
+
+    Returns
+    -------
+    pyvista.UnstructuredGrid
+        Solid sphere mesh.
+
+    See Also
+    --------
+    pyvista.Sphere
```

Comment:
1. Depending on how this renders in the docs (when it's built) it might be better to put a sentence here as an explanation, i.e. that `Sphere` is just the outer shell as a polydata.
2. I strongly recommend cross-referencing the new object in a `See Also` of `Sphere`, that's the much more likely direction that people will find when they look for this new object. So it would help a lot with discoverability.

Reply:
1. numpydoc has short phrases for See Also in its style guide.  I don't see a usage with a long description.  I tried to add something but keep it succint.
2. I added into Spher cross ref.

Reply:
1. Good point, I didn't check the style guide (and still waiting for the doc build to check the current look). At least [the guide says](https://numpydoc.readthedocs.io/en/latest/format.html#see-also) "_Functions may be listed without descriptions, and this is preferable if the functionality is clear from the function name_", so in this case I think it's fine to have a short description (due to lack of obviousness).

Reply:
Built docs look great in this regard:

![screenshot from SolidSphere docs "See Also" section](https://github.com/pyvista/pyvista/assets/17914410/ed911046-27a3-4a9c-837b-07e2b943defb)